test/integration/integration_test_fixtures.py

```diff
@@ -113,12 +124,35 @@ def getTestServerRootUri(self, https=False):
     uri = "%s://%s:%s/" % ("https" if https else "http", uri_host, self.test_server.server_port)
     return uri
 
+  def getAllTestServerRootUris(self, https=False):
+    """
+    Utility for getting the list of http://host:port/ that can be used to query the servers we started
+    in setUp()
+    """
+    uri_host = self.server_ip
```

Comment:
(optional) It tends to be better when if<->else branches are explicit. If we set a variable to value "A" and then conditionally change it to value "B", the reader might get confused, not noticing the condition.

How about:

```python
if self.ip_version == IpVersion.IPV6:
  uri_host = ...
else:
  uri_host = ...
```